software/apps/romi_control/states.c

```diff
@@ -106,26 +86,28 @@ outputs_t transition(inputs_t input_state, system_state_t* curr_state) {
 	        }
 	        break;
 	    }
+		case END_TURNING:
       	case TURNING: {
-      		printf("Integration Val: %f\n", input_state.gyro_integration_z_value);
-      		printf("Cur Angle Val: %f\n", curr_state->curr_orientation_angle);
-      		printf("Cur Turn Val: %f\n", curr_state->turn_angle);
-      		curr_state->curr_orientation_angle = input_state.gyro_integration_z_value; 
+      		curr_state->curr_orientation_angle = input_state->gyro_integration_z_value; 
 	        float diff = angle_modulo(curr_state->turn_angle - curr_state->curr_orientation_angle);
-	        if (input_state.button_pressed || !input_state.has_recently_connected) {
+	        if (input_state->button_pressed || !input_state->has_recently_connected) {
 	          	curr_state->state = OFF;
 	        } else if (fabs(diff) < angle_threshold) {
-	        	//transition out of turning
-	        	//transition into driving
-	        	printf("Transitioning to driving!\n");
-	            curr_state->state = DRIVING;
+				if (curr_state->state == END_TURNING) {
```

Comment:
Not a big fan of having END_TURNING and TURNING in the same case statement. I'd rather have them explicitly be different for simplicity and ease of reading.

Reply:
Yeah I thought it'd be controversial. Honestly the amount of repeated code between the two and the shared functionality (the only difference being exit conditions) made me feel the trade off was worth it. Another way to implement this was by using some sort of state variable to switch between exit conditions, but I felt like keeping the END_TURNING switch case using a fall-through was a good compromise b/w cleanliness and readability